09-features.md

```diff
@@ -30,28 +32,23 @@ The Context column decodes as follows:
-| 0/1   | `option_data_loss_protect`        | Requires or supports extra `channel_reestablish` fields   | IN       |                           | [BOLT #2][bolt02-retransmit]                                          |
-| 3     | `initial_routing_sync`            | Sending node needs a complete routing information dump    | I        |                           | [BOLT #7][bolt07-sync]                                                |
+| 0/1   | `option_data_loss_protect`        | ASSUMED                                                   |          |                           |                                                                       |
 | 4/5   | `option_upfront_shutdown_script`  | Commits to a shutdown scriptpubkey when opening channel   | IN       |                           | [BOLT #2][bolt02-open]                                                |
-| 6/7   | `gossip_queries`                  | More sophisticated gossip control                         | IN       |                           | [BOLT #7][bolt07-query]                                               |
-| 8/9   | `var_onion_optin`                 | Requires/supports variable-length routing onion payloads  | IN9      |                           | [Routing Onion Specification][bolt04]                                 |
-| 10/11 | `gossip_queries_ex`               | Gossip queries can include additional information         | IN       | `gossip_queries`          | [BOLT #7][bolt07-query]                                               |
-| 12/13 | `option_static_remotekey`         | Static key for remote output                              | IN       |                           | [BOLT #3](03-transactions.md)                                         |
-| 14/15 | `payment_secret`                  | Node supports `payment_secret` field                      | IN9      | `var_onion_optin`         | [Routing Onion Specification][bolt04]                                 |
+| 6/7   | `gossip_queries`                  | ASSUMED                                                   |          |                           |                                                                       |
+| 8/9   | `var_onion_optin`                 | ASSUMED                                                   |          |                           |                                                                       |
+| 10/11 | `gossip_queries_ex`               | Gossip queries can include additional information         | IN       |                           | [BOLT #7][bolt07-query]                                               |
+| 12/13 | `option_static_remotekey`         | ASSUMED                                                   |          |                           |                                                                       |
```

Comment:
I don't think we should make this one `ASSUMED`, as it removes the option to cleanly deprecate it. If I'm a node that only wants to support `anchor_outputs` (which we'll want soon), I want to be able to explicitly disable the `option_static_remotekey` feature bit from my `node_announcement` and `init`, to let other nodes know that they shouldn't connect to me if they want `option_static_remotekey` channels.

It is a good idea however to remove the dependency between `anchor_outputs` and `option_static_remotekey` (to allow setting only the `anchor_outputs` feature bit without the `option_static_remotekey` feature bit), and it would be useful to update all implementations to accept that.

Reply:
> that I hope can be implemented in ~1/2 years

Someone's feeling _very_ optimistic 😉

My lofty estimate is something more like 4x that, so ~2 years given the rollout that still needs to happen on the Bitcoin p2p network. 

I think everyone just setting it to be required achieves the same goal in practice? Those that don't support it will fail the feature bit connection check. IIRC in the past when we did this, we had some issues with peers that had old channels, but hadn't yet updated their feature bits: new and old were unable to reconnect as they bit check failed even though there was an existing channel open.

Reply:
> I think everyone just setting it to be required achieves the same goal in practice?

I'm not sure I follow, my goal here is to be able to easily **deprecate** `option_static_remotekey`. Setting it to required definitely does not allow that?

Maybe what you mean here is that making `option_static_remotekey` assumed instead of explicitly set lets us deprecate it cleanly? Which somewhat means we should break the feature link between `anchor_outputs` and `option_static_remotekey`?

Reply:
> This PR essentially reduces the basic channel types to anchors and no anchors. So if we want to require anchor channels, then for now we can set bit 22 (require `option_anchors_zero_fee_htlc_tx`).
> 
> But if there's a new channel type in the future that doesn't depend on anchors, and we want to support both anchors and the new type, we won't have a way to signal that.

You would offer both the new bit and the anchor bit if you allow both, surely?

Reply:
> > I think everyone just setting it to be required achieves the same goal in practice?
> 
> I'm not sure I follow, my goal here is to be able to easily **deprecate** `option_static_remotekey`. Setting it to required definitely does not allow that?

It actually does.  You set it required, and ignore the bit in the peer.  We should fix the definition of negotiated, however, to make this clear.  If you set it required, and they keep talking to you, you assume it's negotiated (this is how CLN works, for example).

Reply:
It is a bit convoluted though...deprecating `option_static_remotekey` by not setting the corresponding feature bit anymore would be way more explicit, wouldn't it? But maybe that ship has sailed since we initially created that dependency, and we just need to make sure we don't recreate the same kind of trap for future channel types.